frame/babe/src/lib.rs

```diff
@@ -203,6 +203,10 @@ decl_storage! {
 		/// if per-block initialization has already been called for current block.
 		Initialized get(fn initialized): Option<MaybeRandomness>;
 
+		/// Temporary value (cleared at block finalization) which is `Some` if we
+		/// have generated VRF randomness.
+		AuthorVrfRandomness get(fn author_vrf_randomness): Option<MaybeRandomness>;
```

Comment:
I know that this is not related to this PR, but we're essentially declaring this as `Option<Option<Randomness>>` (same for `Initialized`). Do we really need the double option wrapping or was it just an oversight? Both keys are supposed to be cleared before block execution finishes.

Reply:
Well for `Initialized` the outer `Option` seems to have a specific meaning? See for example
https://github.com/paritytech/substrate/blob/e65a60cf09f76ae6a80aab1f0653eef84a8e80db/frame/babe/src/lib.rs#L464

However for `AuthorVrfRandomness` yes probably the outer `Option` can be removed

Reply:
Yeah right the semantics for `Initialized` require both options since the outer signals that we have done the initialization procedure. For `AuthorVrfRandomness` I think we could do with removing it and just use `MaybeRandomness`.